w3af/plugins/grep/cdn_providers.py

```diff
@@ -0,0 +1,255 @@
+        [".lswcdn.eu", "LeaseWeb CDN"],
+        [".revcn.net", "Rev Software"],
+        [".revdn.net", "Rev Software"],
+        [".caspowa.com", "Caspowa"],
+        [".twimg.com", "Twitter"],
+        [".facebook.com", "Facebook"],
+        [".facebook.net", "Facebook"],
+        [".fbcdn.net", "Facebook"],
+        [".cdninstagram.com", "Facebook"],
+        [".rlcdn.com", "Reapleaf"],
+        [".wp.com", "WordPress"],
+        [".aads1.net", "Aryaka"],
+        [".aads-cn.net", "Aryaka"],
+        [".aads-cng.net", "Aryaka"],
+        [".squixa.net", "section.io"],
+        [".bisongrid.net", "Bison Grid"],
+        [".cdn.gocache.net", "GoCache"],
+        [".hiberniacdn.com", "HiberniaCDN"],
+        [".cdntel.net", "Telenor"],
+        [".raxcdn.com", "Rackspace"],
+        [".unicorncdn.net", "UnicornCDN"],
+        [".optimalcdn.com", "Optimal CDN"],
+        [".kinxcdn.com", "KINX CDN"],
+        [".kinxcdn.net", "KINX CDN"],
+        [".stackpathdns.com", "StackPath"],
+        [".hosting4cdn.com", "Hosting4CDN"],
+        [".netlify.com", "Netlify"],
+        [".b-cdn.net", "BunnyCDN"],
+        [".pix-cdn.org", "Advanced Hosters CDN"],
+        [".roast.io", "Roast.io"],
+        [".cdnvideo.ru", "CDNvideo"],
+        [".cdnvideo.net", "CDNvideo"],
+        [".trbcdn.ru", "TRBCDN"],
+        [".cedexis.net", "Cedexis"],
+        [".streamprovider.net", "Rocket CDN"],
+    ]
+
+    def grep(self, request, response):
+        """
+        Check if responses are hosted by CDN provider
+        """
+        headers = response.get_headers()
+        for cdn_header in self.cdn_headers:
+            cdn_header_name = cdn_header[0]
+            if cdn_header_name in headers:
+                if cdn_header[1] == headers[cdn_header_name]:
+                    self._detect_cdn(response, cdn_header[2])
+
+        url = request.get_url()
+        domain = url.get_domain()
+        for cdn_domain in self.cdn_domains:
+            if cdn_domain[0] in domain:
+                self._detect_cdn(response, cdn_domain[1])
+
+    def _detect_cdn(self, response, provider_name):
+        """
+        Save grep result to knowledge base
+        """
+        description = 'The URL {} is served using CDN provider: {}'.format(
+            response.get_url(),
+            provider_name
+        )
+        info = Info(
+            'Content Delivery Network Provider detected',
+            description,
+            response.id,
+            self.get_name()
+        )
+        info[CDNProvidersInfoSet.ITAG] = provider_name
+        info.set_url(response.get_url())
+        self.kb_append_uniq_group(
+            self,
+            'cdn_providers',
+            info,
+            group_klass=CDNProvidersInfoSet
+        )
+
+    def get_long_desc(self):
+        """
+        :return: A DETAILED description of the plugin functions and features.
+        """
+        return """
+        Check CDN (Content Delivery Network) providers used by the website.
```

Comment:
`Identify any CDN (Content Delivery Network) providers used by the target website`